### PR TITLE
feat: open source website

### PR DESCRIPTION
**Pull request by Ehesp**
This PR moves the currently private source code for the rnfirebase.io website to the public repository.

### CODE REVIEW

File: docs/messaging/usage/index.md [hunk 1]
```diff
@@ -277,7 +277,7 @@ AppRegistry.registerComponent('app', () => HeadlessCheck);
 
 To inject a `isHeadless` prop into your app, please update your `AppDelegate.m` file as instructed below:
  
-```obj-c
+```objectivec
```
Comment (mikehardy) [inline docs/messaging/usage/index.md:280]: duly noted on that one

File: website/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+First, install the dependencies:
+
+```bash
+yarn
+```
+
+Run the website in development mode:
+
+```bash
+npm run develop
```
Comment (mikehardy) [inline website/README.md:19]: yarn in one spot, npm in another?
```suggestion
yarn develop
```

File: website/gatsby-config.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    {
+      resolve: `gatsby-plugin-sitemap`,
+      options: {
+        exclude: ['/about', 'about'],
+      },
+    },
+    {
+      resolve: `gatsby-plugin-google-tagmanager`,
+      options: {
+        id: `GTM-K2S593S`,
```
Comment (mikehardy) [inline website/gatsby-config.js:20]: Just notifying an identifier. I don't believe this is a problem though, unless there is some sort of abusive hit misattribution?

Comment (Ehesp) [reply website/gatsby-config.js:20]: Nah, it gets added as a script in the HTML anyway so no issues.

**Review by Salakar — APPROVED** [review]

**Pull request merged**